model/config.go

```diff
@@ -61,34 +44,68 @@ func (k Key) String() string {
 // Object references supplied and returned from this interface should be
 // treated as read-only. Modifying them violates thread-safety.
 type ConfigRegistry interface {
-	// Get retrieves a configuration element, bool indicates existence
-	Get(key Key) (proto.Message, bool)
+	// Get retrieves a configuration element.
+	Get(kind, key string) (config proto.Message, exists bool, revision Revision)
 
-	// List returns objects for a kind in a namespace.
-	// Use namespace "" to list resources across namespaces
-	List(kind string, namespace string) (map[Key]proto.Message, error)
+	// List returns objects by type indexed by the key
+	List(kind string) (map[string]proto.Message, error)
```

Comment:
agreed about List. Get in the past won't work since etcd compactifies old revisions. It's not meant to be a linear sequence of revisions since there's no common API to query a range across various KV backends.